package/kernel/linux/modules/fs.mk

```
@@ -332,6 +348,45 @@ endef
 
 $(eval $(call KernelPackage,fs-jfs))
 
+
+define KernelPackage/fs-ksmbd
+  SUBMENU:=$(FS_MENU)
+  TITLE:=SMB kernel server support
+  PROVIDES:=kmod-fs-ksmbd
+  CONFLICTS:=kmod-fs-ksmbd
+  DEPENDS:= \
+	    @!LINUX_5_10 \
+	    +LINUX_5_15:kmod-fs-smbfs-common \
```

Please make this dependency unconditionally, we only support kernel 5.15.
```suggestion
	    +kmod-fs-smbfs-common \
```

---

Done. I think this is what you wanted.

---

I read it as: because we depend on `@!LINUX_5_10` beforehand, no need for the `LINUX_5_15` next line, but what you have done _might_ work as is, because `kmod-fs-smbfs-common` has a !5.10 depend.
Also, if we are switching external for in-tree, we should add the 5.10 dependency on external before we add the in-tree package, and you can cut the PROVIDES and CONFLICTS on the in-tree.

---

Yeah. The @!LINUX_5_10 is an implicit dependency.

---

Still getting these errors:

```
tmp/.config-package.in:10922:error: recursive dependency detected!
tmp/.config-package.in:10922:	symbol PACKAGE_kmod-fs-ksmbd is selected by PACKAGE_ksmbd-server
tmp/.config-package.in:90226:	symbol PACKAGE_ksmbd-server depends on PACKAGE_kmod-fs-ksmbd
For a resolution refer to Documentation/kbuild/kconfig-language.rst
subsection "Kconfig recursive dependency limitations"

tmp/.config-package.in:56063:error: recursive dependency detected!
tmp/.config-package.in:56063:	symbol PACKAGE_luci-app-ksmbd depends on PACKAGE_luci-app-ksmbd
For a resolution refer to Documentation/kbuild/kconfig-language.rst
subsection "Kconfig recursive dependency limitations"

```